front/src/components/app/index.js

```diff
@@ -9,11 +12,17 @@ class App extends Component {
             <BrowserRouter>
                 <Switch>
                     <Route path="/student-registration/:token" component={StudentRegistrationPage} />
+                    <Route path="/teachers" exact component={TeachersPage} />
+                    <Route path="/teachers/create"
+                       render={() => {
+                        this.props.dispatch(showTeacherRegisteredModal())
+                         return <TeachersPage />
+                       }}/>
                     <Route path="/login" component={LoginPage} />
                 </Switch>
             </BrowserRouter>
         )
     }
 }
 
-export default App;
\ No newline at end of file
+export default connect()(App);
```

Comment:
Why do we need `connect` here without arguments

Reply:
Added mapDispatchToProps

Reply:
I removed this logic at all. It crashes the app in case of adding some queries for pagination to URL